src/cadenza.js

```diff
@@ -195,6 +198,22 @@ export class CadenzaClient {
     return this.#iframeElement;
   }
 
+  get #isEmbeddedInCadenza() {
+    return this.#baseUrl === undefined;
+  }
+
+  get #targetWindow() {
+    if (this.#isEmbeddedInCadenza) {
+      assert(
+        !(window.parent === window && window.opener === null),
+        'Cannot find parent Cadenza window',
+      );
+      return window.parent;
+    } else {
+      return /** @type {WindowProxy} */ (this.#requiredIframe.contentWindow);
+    }
+  }
```

Comment:
* Here you're not interested in the URL, so I'd also not check for it.
* `window.parent` is always defined, but it's the same as `window` if there is no parent window.
* If there's an opener, it should also be used.

Proposal:
```javascript
  get #targetWindow() {
    if (this.#iframe) {
      return /** @type {WindowProxy} */ (this.#requiredIframe.contentWindow);
    }

    const targetWindow = window.opener ?? window.parent;
    assert(targetWindow !== window, 'Cannot find target window');
    return targetWindow;
  }
```

Reply:
I added all of the suggestions from comments.